src/azure-cli/azure/cli/__main__.py

```diff
@@ -118,6 +119,11 @@ def cli_main(cli, args):
         logger.warning("Auto upgrade failed. %s", str(ex))
         telemetry.set_exception(ex, fault_type='auto-upgrade-failed')
 
+    try:
+        prompt_survey_message(az_cli)
+    except Exception as ex:  # pylint: disable=broad-except
+        logger.debug("Intercept survey prompt failed. %s", str(ex))
+
```

Comment:
If the command failed in an error, will it prompt survey message?

Reply:
Yes. I have checked with PM if we can prompt survey message only when command execution succeeds. Words from PM

Damien:
>  I feel that we'd be introducing a positive bias by avoiding survey on errors so I am not in favor of doing so.

Pallavi
> I agree with Damien Caro, this will introduce bias. Prompt on errors will get us ton of good data to help us figure out ways to help customers avoid errors, if they give more context and feedback about them.